collections/slice.go

```diff
@@ -31,3 +31,9 @@ func ToMap[TKey comparable, TValue any](slice []TValue, keyFn func(TValue) TKey)
 	}
 	return result
 }
+
+func Reverse[T any](input []T) {
+	for ii, jj := 0, len(input)-1; ii < jj; ii, jj = ii+1, jj-1 {
+		input[ii], input[jj] = input[jj], input[ii]
+	}
+}
```

Comment:
Always safer to create a new output slice than to have side effects on the input. The caller may not expect their input to be mutated internally. The memory overhead of performing the copy is fine since we'll rarely use this on large slices

```suggestion
func Reverse[T any](input []T) []T {
	output := make([]T, len(input)
	for ii := 0, len(input)-1; ii = ii+1 {
		output[len(input) - 1 - ii] = input[ii]
	}
	return output
}
```